### PR TITLE
Allow test frameworks to set context to use.

### PR DESCRIPTION
**Pull request by rwjblue**
Supports efforts in https://github.com/rwjblue/ember-qunit/pull/237.

### CODE REVIEW

File: lib/ember-test-helpers/abstract-test-module.js [hunk 1]
```diff
@@ -100,12 +104,26 @@ export default Klass.extend({
...
+
+    merge(context, {
       dispatcher: null,
       inject: {}
-    }, options);
+    });
+    merge(context, options);
+
+    setContext(context);
+    this.context = context;
```
Comment (Turbo87) [inline lib/ember-test-helpers/abstract-test-module.js]: this line seems unnecessary given the line before

Comment (rwjblue) [inline lib/ember-test-helpers/abstract-test-module.js]: The line before doesn't set `this.context`

File: lib/ember-test-helpers/abstract-test-module.js [hunk 2]
```diff
@@ -100,12 +104,26 @@ export default Klass.extend({
...
+
+  setContext(context) {
+    this.context = context;
+  },
+
+  getContext() {
+    if (this.context) { return this.context; }
 
-    setContext(config);
+    return this.context = getContext() || {};
```
Comment (Turbo87) [inline lib/ember-test-helpers/abstract-test-module.js]: can we make this:

```js
if (!this.context) {
  this.context = getContext() || {};
}
return this.context;
```

File: lib/ember-test-helpers/abstract-test-module.js (hunk 1 above)
Comment (Turbo87) [inline lib/ember-test-helpers/abstract-test-module.js]: ah crap, you're right... missed that the `this` was missing... this is not Java... 😉

File: lib/ember-test-helpers/abstract-test-module.js (hunk 2 above)
Comment (rwjblue) [inline lib/ember-test-helpers/abstract-test-module.js]: Sure, seems functionally the same.  Out of curiosity what makes you prefer that pattern over what I did originally?

Comment (Turbo87) [inline lib/ember-test-helpers/abstract-test-module.js]: it doesn't have the assignment in the return statement. you could make it two statements, but I guess the other proposal is shorter

**Pull request merged**